uaa_bot/bot.py

```diff
@@ -115,3 +115,56 @@ def notify_deactivation_in_10_days(self) -> dict:
             80, "Account of deactivations in 10 days", "account_expires_in_10_days"
         )
         return summary
+
+    def get_all_user_last_logon(
+        self,
+        days_ago: int = 0,
+        days_range: int = 365,
+        summary_title: str = "List Users",
+        start_of_day: int = None,
+        end_of_day: int = None,
+        params: dict = {},
+    ) -> dict:
+        """
+        Gets list of users and their last logon info
+        """
+        users = []
+        uaac = UAAClient(uaa_config=self.uaa_config)
+        uaac.authenticate()
+        if start_of_day and end_of_day:
+            response = uaac.list_users_last_logon(
+                start_of_day=start_of_day, end_of_day=end_of_day, params=params
+            )
+        else:
+            response = uaac.list_users_last_logon(
+                days_ago=days_ago, days_range=days_range, params=params
+            )
+        resources = response.get("resources")
+
+        # Get user with their last logon info
+        for user in resources:
+            user_email = user.get("userName")
+            user_guid = user.get("id")
+            user_last_logon = time.ctime(user.get("lastLogonTime") / 1000)
+            user_active = user.get("active")
+            users.append(
+                {
+                    "user_email": user_email,
+                    "user_guid": user_guid,
+                    "user_last_logon": user_last_logon,
+                    "activated_user": user_active,
+                }
+            )
+
+        # Create and return summary of users' last logon
+        summary = self._summary_response_with_last_logon(summary_title, users)
+        return summary
+
+    def _summary_response_with_last_logon(self, title: str, users: list = []) -> dict:
```

Comment:
feeding the `title` through seems unnecessary - it's not doing any harm, but it could be confusing if we want to reuse parts of this later